nio/client/async_client.py

```diff
@@ -1388,9 +1387,14 @@ async def sync_forever(
                     task.cancel()
 
                 raise
+        self._stop_sync_forever = False
 
     def stop_sync_forever(self):
-        """Request that the `sync_forever` loop exits gracefully.
+        """Request that a running `sync_forever` loop exits gracefully or the next call to `sync_forever()` returns
+        immediately without doing any sync.
+
+        If a `sync_forever` loop was running the loop exits and the `_stop_sync_forever` flag is reset.
```

Comment:
Docs are good, but it'll be better not to reference specific implementation details since:
1. They're simple enough that someone who cares/needs to know can inspect the code directly
2. In case the implementation needs to change, the documentation itself does not need to.
3. Slightly discourages people from attempting to use `self._stop_sync_forever` directly in ways that may not be supported forever (re number 2)
```suggestion
        As `sync_forever` fully shuts down, an internal flag will be reset, allowing `sync_forever` to be called again without shutting down.
```